src/longhorn_rpc_client.c

```diff
@@ -27,13 +27,13 @@ int send_request(struct lh_client_conn *conn, struct Message *req) {
         int rc = 0;
 
         pthread_mutex_lock(&conn->mutex);
-        rc = send_msg(conn->fd, req);
+        rc = send_msg(conn->fd, req, conn->request_header, conn->header_size);
         pthread_mutex_unlock(&conn->mutex);
         return rc;
 }
 
 int receive_response(struct lh_client_conn *conn, struct Message *resp) {
-        return receive_msg(conn->fd, resp);
+        return receive_msg(conn->fd, resp, conn->response_header, conn->header_size);
```

Comment:
Unlike `send_request `, there is no lock protecting `conn->response_header`. Is that OK?

Reply:
Or should/can we malloc the header in `process_request` or `response_process`

Reply:
> Unlike send_request , there is no lock protecting conn->response_header. Is that OK?

`response_process` is single thread, so it is feasible. @keithalucas is it correct?

> Or should/can we malloc the header along with the req/resp in process_request/response_process?

malloc in `response_process` is ok, but `process_request` is called while receiving each request.